### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by ngiotta** [RainbowMiner/RainbowMiner#2698]
In the later versions of Rainbowminer, there seems to be a massive memory leak.  It's using over 3.2GB RAM.

![Screenshot 2024-03-02 221419](https://github.com/RainbowMiner/RainbowMiner/assets/2933262/6fb798dd-f6ee-4cb1-b6ec-a35807213416)

**Comment by RainbowMiner** [comment]
Which version of the powershell core are you using?

**Comment by ngiotta** [comment]
Good to know.  I have 7.4.1.5.  I will try downgrading.  Thanks.

**Comment by UselessGuru** [comment]
@ngiotta 
Any better with 7.2.x?

**Comment by ngiotta** [comment]
> @ngiotta Any better with 7.2.x?

Nope.  I uninstalled PS and installed 7.2.18 and the memory leak remains.  Seems to only be on my W11 boxes.      I'm just going to run Rainbow on those machines, because it's so bad, they start crashing.

**Comment by RainbowMiner** [comment]
That's strange - all my Win10 systems with Powershell Core v7.2.18 run stable at around 250MByte memory consumption. Could you please upload a Debug file? Maybe there is a faulty pool module.

**Comment by RainbowMiner** [comment]
> Could you implement a workaround, that the PowerShell script restarts itself, if it reaches a configured memory threshold?

That is in fact a very good idea. I will add this workaround until we find the culprit.

**Comment by RainbowMiner** [comment]
> Can I send you the logs via mail or Discord for privacy?

Sure: send it to info@rbminer.net

**Comment by UselessGuru** [comment]
> That's strange - all my Win10 systems with Powershell Core v7.2.18 run stable at around 250MByte memory consumption

I am not actively using RBM, but I can confirm that all PWSH versions I have tested so far have a memory leak (approx 300MB/day) when running UG-Miner which uses a lot of similar code.

**Comment by RoyCRoderickIII** [comment]
I dont think its RBM personally, i have both a very crappy windows laptop running Win 10 pro, and my pi running ubuntu 22.04 and neither get above 6GB of ram.usage running daily. So it has to be a miner issue, seeing i dont GPU mine at all.

**Comment by UselessGuru** [comment]
>  So it has to be a miner issue, seeing i dont GPU mine at all.

Nope, I even see the mem leak when just running UG-Miner running in 'dry run' mode (constantly querying pools & building miner objects to get the most profitable miners, but not actually running miners).

**Comment by RoyCRoderickIII** [comment]
> >  So it has to be a miner issue, seeing i dont GPU mine at all.
> 
> Nope, I even see the mem leak when just running UG-Miner running in 'dry run' mode (constantly querying pools & building miner objects to get the most profitable miners, but not actually running miners).

Thats kinda my point, if it was RBM in general, all of us would be experiencing it, or just the window versions etc. it must be either a miner issue or the way RBM opens/closes the miner or miner version.

**Comment by UselessGuru** [comment]
> it must be either a miner issue or the way RBM opens/closes the miner or miner version.

Not in my case. The mem leak is happing regardless of miners getting launched or not.

**Comment by mr-manuel** [comment]
I can confirm that. Only the managing process crashes, the miner itself continues to mine.
